action.go

```diff
@@ -434,3 +435,19 @@ func (a Action) UsageF(f func() string) Action {
 		return a
 	})
 }
+
+func (a Action) Match(m match.Match) Action {
+	return a.MatchF(func() match.Match {
+		return m
+	})
+}
+
+func (a Action) MatchF(f func() match.Match) Action { // TODO should UsageF, MatchF .. get passed context as default?
```

Comment:
Breaking the API with contexts is going to have a cost.

Rather you might do the classic Go option pattern:
```
type Options func(o *internalOpts) 
```

with functions now looking like:
```
func TagF(func(val string) string, opts ...Options)

func UsageF(func(val string) string, opts ...Options)
```

That is quite useful: you can expand the Options API, more easily shrink it if you really need to (by slowly deprecating options), and you can then do anything within your code...